mlflow/server/js/src/experiment-tracking/components/CompareRunView.js

```diff
@@ -68,6 +117,12 @@ export class CompareRunView extends Component {
       <Link to={Routes.getExperimentPageRoute(experimentId)}>{experiment.getName()}</Link>,
       title,
     ];
+
+    function onCollapsibleSectionChanged(blockClass, hideBlock) {
+      const blockElem = document.querySelectorAll(`.compare-table .${blockClass}`)[0];
+      blockElem.style.display = hideBlock ? 'none' : 'block';
+    }
```

Comment:
Controlling visibility like this is not straightforward. Can we define a state, and toggle that state with a callback?

Reply:
I know setting a state is a more react-js code way. But seems it is not as efficient here. After state changed, the render will be called again to generate all virtual doms, and then we need call `adjustTableColumnWidth` method again to adjust all table cells width again. In this scenario directly setting the real DOM style is a more straightforward way.